erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -306,13 +307,27 @@ def get_pending_data(self, data_type, s_date=None, item_name=None):
 					return schedule.name
 
 @frappe.whitelist()
-def update_serial_nos(s_id):
-	serial_nos = frappe.db.get_value('Maintenance Schedule Detail', s_id, 'serial_no')
+def update_serial_nos(s_id, item):
+	if isinstance(item, str):
+		item = json.loads(item)
+		item = frappe._dict(item)
+
+	serial_nos = []
+	if s_id:
+		serial_nos = frappe.db.get_value('Maintenance Schedule Detail', s_id, 'serial_no')
+
+	# for old docs which were already saved.
+	elif item.get('prevdoc_docname'):
```

Comment:
This can't be patched to handle changes in code? So we don't have to add code like this.

Reply:
I had removed the mapping to these fields. So the patch could be copying this prevdoc_docname(if present) to the form level field(both are storing the maintenance schedule name)? I could just use the maintenance schedule name while fetching the serial nos but then I wouldn't be covering cases where the user could add duplicate items with a single serial no

![serial](https://user-images.githubusercontent.com/43572428/146127174-97f1eef3-8bb3-4f4e-8476-11f589cab84d.gif)

Currently it uses the linked row id from the schedules table in Maintenance Schedule, can't get in old docs since this was never linked. Should I just ignore this case and use the name?

Reply:
Or could just not filter the serials for old docs(old behaviour).

Reply:
Fetching serials with maintenance Schedule name now, wrote a patch to update the maintenance schedule field.

Reply:
What type of tests should I write for this?